.github/workflows/npm-build.yml

```diff
@@ -0,0 +1,41 @@
+name: Lock npm packages
+on: 
+  push:
+    paths:
+      - 'yarn.lock'
+
+jobs:
+  build:
+    name: Build & Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          registry-url: https://registry.npmjs.org/
+
+      - name: Remove old NPM lockfile
+        run: |
+          cd packages/hoprd
+          rm -f package-lock.json
+      - name: Install dependencies and create new NPM lockfile
+        run: |
+          cd packages/hoprd
+          npm install
+          npm run-script build
+      - name: Start with npm and check if it fails
+        timeout-minutes: 1
+        run: |
+          cd packages/hoprd
+          npm run-script start:dryRun
```

Comment:
I don't think a running hoprd is within the scope of this workflow

Reply:
It is. Running `hoprd` is necessary because we need to *execute* package resolution.

Reply:
does that mean that npm's lock file may not work as we expect?

Reply:
Yes, exactly. This is the reason why it's tested afterwards.

Reply:
if it doesn't work as expected then we will have to find another solution & submit an issue to them, because if it doesn't work as expected then our CI will fail with this and block us, how do you know it doesn't always work?

Reply:
The issue is on our side because we are using `yarn` to develop and `npm` when deploying. The whole idea is to freeze package versions, check if this works and fail if manual work is required